### PR TITLE
Replace old issue templates with new yml-based issue templates

### LINKED ISSUE #25124: Replace the old .md based issue templates with the new .yml based issue templates
## Reason

Github has new .yml based issue templates, which IMHO looks much nicer plus it offers field validations.

## Design

Replace the old issue templates with new yml templates.

### PR DESCRIPTION
**Pull request by hugary1995**
close #25124

The new .yml issue templates are much more user-friendly, IMHO. It has a few advantages compared to the old .md based templates:
- Allows for textareas.
- Allows for validation (some fields are required)
- I reserved empty templates, so you can create an issue without templates if absolutely necessary.
- There is a redirect link at the bottom pointing users to the discussion page if they want to ask questions.

### CODE REVIEW

**Comment by loganharbour** [comment]
Thanks @hugary1995; is there not an easy way to test this?

**Comment by hugary1995** [comment]
If you view the file in the browser, I think GitHub will automatically render it. But that doesn't look the same as the real thing...

The easiest way I can think of is to test this in a dummy repo.

**Comment by hugary1995** [comment]
Here, take a look: https://github.com/hugary1995/test-issue-template

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+name: Bug Report
+description: Report a bug that is preventing proper operation of the framework or an application
+
```
Comment (cticenhour) [inline .github/ISSUE_TEMPLATE/bug-report.yml:3]:
```suggestion
labels: 'T: defect, P: normal'

```

**Review by cticenhour — CHANGES_REQUESTED** [review]
I like this idea. The new form in the test repository looks really nice!

Only one suggestion: I noticed that the labels were missing from the bug report. Otherwise it seems like everything else we require is there.

**Comment by hugary1995** [comment]
I just committed your suggestion, assuming you've tested it. If not I can make the change in the test repo.

**Comment by cticenhour** [comment]
Re-opening, as we'll need to upgrade to these eventually. Assigning myself, so I can explore them and give a proper review.

File: .github/ISSUE_TEMPLATE/feature-request.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      Why do you need this feature or enhancement?
+  validations:
+    required: true
+- type: textarea
+  attributes:
+    label: Design
+    description: |
+      A concise description (design) of the feature or enhancement.
+  validations:
+    required: false
```
Comment (cticenhour) [inline .github/ISSUE_TEMPLATE/feature-request.yml:19]:
```suggestion
    required: true
```

File: .github/ISSUE_TEMPLATE/feature-request.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+- type: textarea
+  attributes:
+    label: Impact
+    description: |
+      Will the enhancement change existing APIs or add something new?
+  validations:
+    required: false
```
Comment (cticenhour) [inline .github/ISSUE_TEMPLATE/feature-request.yml:26]:
```suggestion
    required: true
```

File: .github/ISSUE_TEMPLATE/config.yml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+blank_issues_enabled: true
+contact_links:
+  - name: Questions
+    url: https://github.com/idaholab/moose/discussions
+    about: Ask questions and discuss with other MOOSE community members
```
Comment (cticenhour) [inline .github/ISSUE_TEMPLATE/config.yml:5]:
```suggestion
    about: Ask general questions about MOOSE and discuss with other MOOSE community members.
```

**Review by cticenhour — CHANGES_REQUESTED** [review]
Need to make more fields required for SQA for the Feature Request. Also made a suggested adjustment to the Questions link description.

**Comment by cticenhour** [comment]
@hugary1995 Do you mind squashing those last three commits? Just in case you didn't know, you can batch commit review suggestions by going into the "Files changed" tab and handling them there.

**Comment by hugary1995** [comment]
Good to know!

**Review by cticenhour — APPROVED** [review]

**Review by lindsayad — APPROVED** [review]

**Pull request merged**